features/environment.feature

```diff
@@ -0,0 +1,25 @@
+Feature: Git Town performs correctly depending on the environment
+
+  Scenario Outline: Git Town commands run outside of a Git repository
+    Given I'm currently not in a git repository
+    When I run `<COMMAND>` while allowing errors
+    Then I <DO_OR_DONT> see "This is not a git repository."
+    And I don't see "fatal: Not a git repository"
+
+  Examples:
+    | COMMAND                       | DO_OR_DONT |
+    | git town                      | don't      |
+    | git town help                 | don't      |
+    | git town version              | don't      |
+    | git town config               | do         |
+    | git town main-branch          | do         |
+    | git town non-feature-branches | do         |
+    | git extract                   | do         |
+    | git hack                      | do         |
+    | git kill                      | do         |
+    | git pr                        | do         |
+    | git prune-branches            | do         |
+    | git repo                      | do         |
+    | git ship                      | do         |
+    | git sync                      | do         |
+    | git sync-fork                 | do         |
```

Comment:
How about updating this to

```cucumber
Feature: Git Town performs correctly depending on the environment

  Scenario Outline: Git Town commands run outside of a Git repository
    Given I'm currently not in a git repository
    When I run `<COMMAND>` while allowing errors
    Then I see "This is not a git repository."
    And I don't see "fatal: Not a git repository"

    Examples:
    | COMMAND                       | 
    | git town config               | 
    | git town main-branch          | 
    | git town non-feature-branches |
    | git extract                   | 
    | git hack                      | 
    | git kill                      | 
    | git pr                        | 
    | git prune-branches            | 
    | git repo                      | 
    | git ship                      |
    | git sync                      |
    | git sync-fork                 |
```

And adding tests verifying that `git town`, `git town help`, and `git version` work as expected outside of a git repository to their respective features (features/git-town/help, feature/git-town/version.feature).